netbox/netbox/search/backends.py

```diff
@@ -120,8 +121,14 @@ def search(self, value, user=None, object_types=None, lookup=DEFAULT_LOOKUP_TYPE
             except (AddrFormatError, ValueError):
                 pass
 
+        # Exclude any records which refer to an instance of a model that's no longer installed. This
+        # can happen when a plugin is removed but its data remains in the database, for example.
+        content_type_ids = set(
+            ct.pk for ct in ContentType.objects.get_for_models(*apps.get_models()).values()
+        )
+
         # Construct the base queryset to retrieve matching results
-        queryset = CachedValue.objects.filter(query_filter).annotate(
+        queryset = CachedValue.objects.filter(object_type_id__in=content_type_ids).filter(query_filter).annotate(
```

Comment:
This introduces significant overhead to the query. We should explore options for a more elegant approach, which will probably entail prescribing a process for clearing cached values for plugins that have been removed.